misk/src/main/kotlin/misk/tasks/RepeatedTaskQueue.kt

```diff
@@ -31,22 +37,30 @@ class RepeatedTaskQueue @VisibleForTesting internal constructor(
   private val taskExecutor: ExecutorService,
   private val dispatchExecutor: Executor?, // visible internally for testing only
   private val pendingTasks: BlockingQueue<DelayedTask>, // visible internally for testing only
+  // TODO(rhall): make required when all callers are using the Factory
+  metrics: Metrics?,
   private val config: RepeatedTaskQueueConfig = RepeatedTaskQueueConfig()
 
 ) : AbstractExecutionThreadService() {
   /**
    * Creates a [RepeatedTaskQueue] backed by a real [DelayQueue], with tasks dequeued on the
    * service background thread and executed via the provided [ExecutorService]
    */
+  @Deprecated("use the factory instead", replaceWith = ReplaceWith("RepeatedTaskQueueFactory"))
   constructor(
     name: String,
     clock: Clock,
     taskExecutor: ExecutorService,
     config: RepeatedTaskQueueConfig = RepeatedTaskQueueConfig()
   ) :
-      this(name, clock, taskExecutor, null, DelayQueue<DelayedTask>(), config)
+      this(name, clock, taskExecutor, null, DelayQueue<DelayedTask>(), null, config)
 
   private val running = AtomicBoolean(false)
+  internal val taskLatency = metrics?.histogram(
+      "task_latency",
```

Comment:
This isn’t latency, it’s duration. task_duration

Also I wonder if there should be a prefix: task_queue_task_duration, since Prometheus metrics aren’t namespaced

Reply:
I wonder if we should have prefixed all misk metrics with `misk_`. oh well